osu.Game.Rulesets.Mania/UI/ManiaPlayfield.cs

```diff
@@ -82,6 +85,9 @@ public ManiaPlayfield(List<StageDefinition> stageDefinitions)
             }
         }
 
+        [Pure]
```

Comment:
This annotation is supposed to assuage doubts that may pop up from the "virtual call in ctor" hint-level inspections resulting from this change.